releasenotes/notes/operation-abstract-base-class-c5efe020aa9caf46.yaml

```diff
@@ -0,0 +1,15 @@
+---
+
+features:
+  - |
+    A new :class:`.Operation` base class for objects that can be added to a :class:`.QuantumCircuit`.
+    These objects include :class:`.Gate`, :class:`.Reset`, :class:`.Barrier`, :class:.Measure`,
+    and operators such as :class:`.Clifford`.
+  - |
+    A Clifford gate is now added to a quantum circuit as an :class:`.Operation`, without first
+    synthesizing a subcircuit implementing this Clifford gate. The actual synthesis is postponed
+    to a later transpilation pass.
+  - |
+    Added a new transpiler pass :class:`.OptimizeCliffords` that collects blocks of consecutive
+    Clifford gates in a circuit, and replaces each block with a single Clifford object.
```

Comment:
Thanks. I was somehow hesitant to already include `OptimizeCliffords` into preset pass managers. Maybe when more higher-level objects become available and when the pass becomes a little bit stronger?

Regarding the last point, the pass is currently restricted to Cliffords with identical qubits (viewed as ordered lists). It is likely that one can also combine Cliffords when the sets of qubits are equal (see https://github.com/Qiskit/qiskit-terra/pull/7966#discussion_r869037645), or when the sets of qubits intersect, but this requires a bit more work.

Reply:
Yeah, that's fine I think we can start with this as a standalone pass. I only mentioned that to say it was important we document how it's used since it won't be done automatically under the covers in a preset pass manager so people will be looking to the docs to understand how to use the `OptimizeCliffords` pass.